InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -0,0 +1,85 @@
+/* global instantsearch */
+
+function getCategorySlug(name) {
+  return name.replace(/\s/g, '-');
+}
+
+function getCategoryName(slug) {
+  return slug.replace(/-/g, ' ');
+}
+
+const router = instantsearch.routers.history({
+  windowTitle({ category, q }) {
+    const query = q ? `Results for "${q}"` : 'Search';
+
+    if (category) {
+      return [category, query].join(' – ');
+    }
+
+    return query;
+  },
+  createURL({ qsModule, routeState, location }) {
+    const { protocol, hostname, port = '', hash } = location;
+    const portWithPrefix = port === '' ? '' : `:${port}`;
+    const pathname = routeState.category
+      ? `/${getCategorySlug(routeState.category)}/`
+      : '';
+
+    const searchState = {};
+    if (routeState.q) searchState.q = encodeURIComponent(routeState.q);
+    if (routeState.p !== 1) searchState.p = routeState.p;
+    if (routeState.brands) searchState.brands = routeState.brands;
```

Comment:
Is there a reason why we don't encode the brand values? They might also contain space or weird character.

Reply:
Nope, I added that.